test/dev.test.mjs

```diff
@@ -0,0 +1,41 @@
+import assert from "node:assert";
+import { afterEach, beforeEach, describe, it } from "node:test";
+import { devServerURL, launchDemoDevServer } from "./util.mjs";
+
+describe("dev server", async () => {
+	let devServerProc;
+	beforeEach(async () => {
+		devServerProc = await launchDemoDevServer();
+	});
+
+	afterEach(() => {
+		// Until nodejs/node#45204 is released in node v18, this won't be called on
+		// test failures, so we'll need to manually wrap tests for now
+		devServerProc.kill();
+	});
+
+	// TODO: Remove this wrapper once nodejs/node#45204 is released
+	const wrap = fn => {
+		return async (...args) => {
+			try {
+				await fn(...args);
+			} finally {
+				devServerProc.kill();
+			}
+		};
+	};
+
+	it(
+		"serves src/main.tsx",
+		wrap(async () => {
+			const mainURL = new URL("/src/main.tsx", devServerURL);
+			const res = await fetch(mainURL);
```

Comment:
THE FUTURE! I'm using the new built-in NodeJS test runner and global fetch!! No dependencies <3

The test runner still has some bugs as mentioned in the code comments above, but still cool for simple situations like this!